obspy/io/sac/arrayio.py

```diff
@@ -0,0 +1,637 @@
+# -*- coding: utf-8 -*-
+"""
+Low-level array interface to the SAC file format.
+
+Functions in this module work directly with numpy arrays that mirror the SAC
+format.  The 'primitives' in this module are the float, int, and string header
+arrays, the float data array, and a header dictionary. Convenience functions
+are provided to convert between header arrays and more user-friendly
+dictionaries.
+
+These read/write routines are very literal; there is almost no value or type
+checking, except for byteorder and header/data array length.  File- and array-
+based checking routines are provided for additional checks where desired.
+
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.utils import native_str
+from future.builtins import *
+
+import sys
+import warnings
+
+import numpy as np
+
+from obspy.core.compatibility import from_buffer
+
+from ..sac import header as HD
+from .util import SacIOError, SacInvalidContentError
+from .util import is_valid_enum_int, is_same_byteorder, byteswap
+
+
+def init_header_arrays(*args):
+    """
+    Initialize arbitrary header arrays.
+
+    Parameters
+    ----------
```

Comment:
:disappointed:  I know reST makes pretty web pages, but docstrings are generally viewed as text, and reST is hideous to look at.